extra/html-extra/HtmlExtension.php

```diff
@@ -124,4 +130,77 @@ public static function htmlCva(array|string $base = [], array $variants = [], ar
     {
         return new Cva($base, $variants, $compoundVariants, $defaultVariant);
     }
+
+    public static function htmlAttrMerge(...$arrays): array
+    {
+        $result = [];
+
+        foreach ($arrays as $argNumber => $array) {
+            if (!$array) {
+                continue;
+            }
+
+            if (!is_iterable($array)) {
+                throw new RuntimeError(sprintf('The "attr_merge" filter only works with arrays or "Traversable", got "%s" for argument %d.', \gettype($array), $argNumber + 1));
+            }
+
+            $array = CoreExtension::toArray($array);
+
+            foreach (['class', 'style', 'data', 'aria'] as $deepMergeKey) {
+                if (isset($array[$deepMergeKey])) {
+                    $value = $array[$deepMergeKey];
+                    unset($array[$deepMergeKey]);
+
+                    if (!is_iterable($value)) {
+                        $value = (array) $value;
+                    }
+
+                    $value = CoreExtension::toArray($value);
+
+                    $result[$deepMergeKey] = array_merge($result[$deepMergeKey] ?? [], $value);
```

Comment:
Can we use `[... ]` when we can ? As it's slightly more performant than array_merge

Reply:
@smnandre Like this? https://github.com/twigphp/Twig/pull/3930/commits/717f872dbb6eb32d19b50009d98fa3aad94fddd3